Issue #91: Update gh pages

ronja-ui opened the pull request:
Updated the look-and-feel of index.html for GitHub pages. 

Changes:
- Improved the HTML structure
- Added external CSS files
- Imported brand typeface as a self-hosting font

Features:
- dark(ish) theme
- 2–4 grid that scales in differents screen sizes

Screenshots:

NOTE: in the actual version, the main branch won't be copy pasted like in the screenshots.

<img width="1631" height="819" alt="Screenshot_20250926_152700" src="https://github.com/user-attachments/assets/f9dd368e-9bf8-47ae-8edd-12bf23af2c4c" />
<img width="972" height="816" alt="Screenshot_20250926_152721" src="https://github.com/user-attachments/assets/fcd52b4f-ee73-46cc-bebc-e98b5a879a19" />

ronja-ui committed:
Update index.html

updated HTML structure

ronja-ui committed:
Create css.css

added basic stylings

ronja-ui committed:
Add files via upload

added roboto mono font

ronja-ui committed:
Rename RobotoMono-Bold.ttf to roboto-mono/RobotoMono-Bold.ttf

relocated

ronja-ui committed:
Rename RobotoMono-BoldItalic.ttf to roboto-mono/RobotoMono-BoldItalic.ttf

relocated

ronja-ui committed:
Rename RobotoMono-Italic.ttf to roboto-mono/RobotoMono-Italic.ttf

relocated

ronja-ui committed:
Rename RobotoMono-Medium.ttf to roboto-mono/RobotoMono-Medium.ttf

relocated

ronja-ui committed:
Rename RobotoMono-MediumItalic.ttf to roboto-mono/RobotoMono-MediumItalic.ttf

relocated

ronja-ui committed:
Rename RobotoMono-Regular.ttf to roboto-mono/RobotoMono-Regular.ttf

relocated

ronja-ui committed:
Rename RobotoMono-SemiBold.ttf to roboto-mono/RobotoMono-SemiBold.ttf

relocated

ronja-ui committed:
Rename RobotoMono-SemiBoldItalic.ttf to roboto-mono/RobotoMono-SemiBoldItalic.ttf

relocated

ronja-ui committed:
Create font.css

defined roboto mono typeface

ronja-ui committed:
Update css.css

included font.css and changed the used typeface

ronja-ui committed:
Update repository.js

revert me later

ronja-ui committed:
Update index.html

added container div around the content

ronja-ui committed:
Update index.html

fixed a brain fart

ronja-ui committed:
Update css.css

turned the content into a flexbox

ronja-ui committed:
Rename css.css to css/css.css

created a new directory for css files

ronja-ui committed:
Rename font.css to css/font.css

relocated

ronja-ui committed:
Update index.html

fixed css link

ronja-ui committed:
Create variables.css

created a separate file for variables

ronja-ui committed:
Update css.css

included the new variables css file

ronja-ui committed:
Update css.css

fixed typo

ronja-ui committed:
Update index.html

added a div container for branches

ronja-ui committed:
Update variables.css

added margin and padding utility values

ronja-ui committed:
Update css.css

adjusted margins by using spacing utilities

ronja-ui committed:
Update css.css

adjusted how branches show in the grid and the content is shown when there's many branches displayed

ronja-ui committed:
Update css.css

added title divider for h1

ronja-ui committed:
Update css.css

made the title divider responsive

ronja-ui committed:
Update css.css

made the grid to scale in different screen sizes

ronja-ui committed:
Update repository.js

reverted earlier commit before creating PR

q22u reviewed (commented):
commit messages were almost identical, tedious to read, good otherwise

ronja-ui commented on css/css.css:
```diff
@@ -74,8 +74,16 @@ h2:last-child {
+  .branches { grid-template-columns: auto auto auto; }
+}
+
+@media screen and (max-width: 600px) {
```
No, because it's the exact point when there's a need for scaling adjustments. I can add a comment to clarify why it's 600px.

q22u replied:
ok

ronja-ui commented on css.css:
```diff
@@ -0,0 +1,46 @@
+  text-transform: uppercase;
+}
+
+h1            { margin-bottom: 48px; }
```
it's currently using a variable so this comment is outdated. However, the sizing utilities could be in rem unit.

q22u replied:
ok

ronja-ui replied:
and done

ronja-ui commented on index.html:
```diff
@@ -1,15 +1,23 @@
+        htmlString += "</div>";
       }
-      htmlString += '</ul>';
+      htmlString += `<h2>Teragrep</h2>`;
```
I would keep it as it is since the main focus shouldn't be in either "mvn_01" or "Teragrep" headings but in the actual content.

q22u replied:
ok

ronja-ui commented on css.css:
```diff
@@ -0,0 +1,46 @@
+body {
+  background-color: var(--dm-grey-600);
+  color: var(--dm-grey-200);
+  font-size: 16px;
```
I'll increase it to 18px.

q22u replied:
ok

ronja-ui replied:
and done

ronja-ui committed:
Update css.css

Increased font size to increase readability and commented code to clarify @media rules.

ronja-ui committed:
Update variables.css

changed spacing utility values to rem unit

1rem = 16px unless a different value is attached for the `<html>` element.
